etfl/io/dict.py

```diff
@@ -55,10 +55,18 @@ def expressed_gene_to_dict(gene):
     obj['rna'] = str(gene.rna)
     obj['peptide'] = str(gene.peptide)
     obj['copy_number'] = str(gene.copy_number)
-    obj['transcribed_by'] = [x.id for x in gene.transcribed_by] \
-        if gene.transcribed_by is not None else None
-    obj['translated_by'] = [x.id for x in gene.translated_by]\
-        if gene.translated_by is not None else None
+    try:
+        obj['transcribed_by'] = [x.id for x in gene.transcribed_by] \
+            if gene.transcribed_by is not None else None
+    except AttributeError:
+        obj['transcribed_by'] = [x for x in gene.transcribed_by] \
+            if gene.transcribed_by is not None else None
+    try:
+        obj['translated_by'] = [x.id for x in gene.translated_by]\
+            if gene.translated_by is not None else None
+    except AttributeError:
+        obj['translated_by'] = [x for x in gene.translated_by] \
+            if gene.translated_by is not None else None            
```

Comment:
I see no method to read that back from the dictionary ?

Reply:
I will do this later ;)